assets/js/modules/analytics-4/datastore/properties.test.js

```diff
@@ -607,6 +611,180 @@ describe( 'modules/analytics-4 properties', () => {
+					registry.select( MODULES_ANALYTICS_4 ).getGoogleTagID()
+				).toEqual( '' );
+
+				expect(
+					registry
+						.select( MODULES_ANALYTICS_4 )
+						.getGoogleTagLastSyncedAtMs()
+				).toEqual( 0 );
+			} );
+
+			it( 'should not execute if Google Tag settings already exist', async () => {
+				enabledFeatures.add( 'gteSupport' );
+				provideModules( registry, [
+					{
+						slug: 'analytics-4',
+						active: true,
+						connected: true,
+					},
+				] );
+
+				registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {
+					measurementID: 'G-1A2BCD346E',
+					googleTagID: 'GT-12345',
+					googleTagLastSyncedAtMs: 1676519625246,
+				} );
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.syncGoogleTagSettings();
+
+				expect(
+					registry.select( MODULES_ANALYTICS_4 ).getGoogleTagID()
+				).toEqual( 'GT-12345' );
+
+				expect(
+					registry
+						.select( MODULES_ANALYTICS_4 )
+						.getGoogleTagLastSyncedAtMs()
+				).toEqual( 1676519625246 );
+			} );
+
+			it( 'should not execute if measurement ID is not set', async () => {
+				enabledFeatures.add( 'gteSupport' );
+				provideModules( registry, [
+					{
+						slug: 'analytics-4',
+						active: true,
+						connected: true,
+					},
+				] );
+
+				registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {
+					measurementID: '',
+					googleTagID: '',
+					googleTagLastSyncedAtMs: 0,
+				} );
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.syncGoogleTagSettings();
+
+				expect(
+					registry.select( MODULES_ANALYTICS_4 ).getGoogleTagID()
+				).toEqual( '' );
+
+				expect(
+					registry
+						.select( MODULES_ANALYTICS_4 )
+						.getGoogleTagLastSyncedAtMs()
+				).toEqual( 0 );
+			} );
+
+			it( 'should not execute if settings were synced less than an hour ago', async () => {
+				enabledFeatures.add( 'gteSupport' );
+				provideModules( registry, [
+					{
+						slug: 'analytics-4',
+						active: true,
+						connected: true,
+					},
+				] );
+
+				registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {
+					measurementID: 'G-1A2BCD346E',
+					googleTagID: '',
+					googleTagLastSyncedAtMs: Date.now() - 1800000, // 30 minutes ago.
+				} );
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.syncGoogleTagSettings();
+
+				expect(
+					registry.select( MODULES_ANALYTICS_4 ).getGoogleTagID()
+				).toEqual( '' );
+			} );
+
+			it( 'dispatches a request to get and populate Google Tag settings', async () => {
+				enabledFeatures.add( 'gteSupport' );
+				provideModules( registry, [
+					{
+						slug: 'analytics-4',
+						active: true,
+						connected: true,
+					},
+				] );
+
+				const measurementID = 'G-1A2BCD346E';
+
+				registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {
+					measurementID,
+					googleTagID: '',
+					googleTagLastSyncedAtMs: 0,
+				} );
+
+				fetchMock.getOnce( googleTagSettingsEndpoint, {
+					body: fixtures.googleTagSettings,
```

Comment:
It would be worth passing a deep copy of `fixtures.googleTagSettings` here, so we can be sure that the code being executed for the test does not inadvertently change the values we're using in the assertions.

```suggestion
					body: cloneDeep( fixtures.googleTagSettings ),
```

Reply:
Updated, thank you!